modules/core/keeper/keeper.go

```diff
@@ -61,7 +63,9 @@ func NewKeeper(
 		panic(fmt.Errorf("cannot initialize IBC keeper: empty scoped keeper"))
 	}
 
-	clientKeeper := clientkeeper.NewKeeper(cdc, key, paramSpace, stakingKeeper, upgradeKeeper)
+	authority := authtypes.NewModuleAddress(govtypes.ModuleName).String()
```

Comment:
q: any reason why this shouldn't be done here? This results in no changes to `app.go` but I'm not completely sure if there's some gotcha.

Reply:
I think the gotcha is that this won't register if the authority address changes by governance vote. Also, this is generating a new address and not using already existing address.

Reply:
I guess the authority field should be passed to the ibc core keeper in app.go and propagated to submodules

Reply:
Yep, address generation is deterministic